File: constant.cpp
#include <vector>
#include <iostream>
#include <chrono>
#include <utility>

void constant(int input_size) {
    ;
}

int main() {
    
    const int MAX_SIZE = 10000000;
    const int INCREMENT = 100000;
    const int MAX_RUNTIME = 10;
    
    int input_size = 0;
    std::vector<std::pair<int, double>> results;

    std::chrono::duration<double> measurment;
    

    while (input_size <= MAX_SIZE && measurment.count() <= MAX_RUNTIME) {
        

        auto start1 = std::chrono::high_resolution_clock::now();
        constant(input_size);
        auto end1 = std::chrono::high_resolution_clock::now();

        auto start2 = std::chrono::high_resolution_clock::now();
        constant(input_size);
        auto end2 = std::chrono::high_resolution_clock::now();

        measurment = ((end1 - start1) + (end2 - start2)/2);

        std::cout << "Round: " << input_size << "  Time: " << measurment.count() << " s" << std::endl;

        results.push_back({input_size, measurment.count()});

        input_size += INCREMENT;

    }

    std::cout << "\n\nComplete\ncsv\n\nx,y" << std::endl;
    for (const auto& i : results) {
        std::cout << i.first << "," << i.second << std::endl;
    }
    
    return 0;
}

File: constant.py
import timeit

def constant(input_size):
    pass


MAX_SIZE = 10000000
INCREMENT = 100000
MAX_RUNTIME = 10

input_size = 0
measurment = 0

results = []

while input_size <= MAX_SIZE and measurment/2 <= MAX_RUNTIME:

    measurment = timeit.timeit(lambda: constant(input_size), number=2)

    print(f"Round: {input_size}  Time: {measurment}  Avg: {measurment/2}")

    results.append([input_size, measurment/2])

    input_size = input_size + INCREMENT

print("\n\nComplete\ncsv\n\nx,y")
for i in results:
    print(*i, sep = ",")